src/tools/cache.ts

```diff
@@ -0,0 +1,88 @@
+import * as crypto from "crypto"
+import { filesystem, GluegunFilesystem } from "gluegun"
+import type { PackagerName } from "./packager"
+
+const lockFile = {
+  yarn: "yarn.lock",
+  pnpm: "pnpm-lock.yaml",
+  npm: "package-lock.json",
+} as const
+
+const MAC: NodeJS.Platform = "darwin"
+const WINDOWS: NodeJS.Platform = "win32"
+const LINUX: NodeJS.Platform = "linux"
+const cachePath = {
+  [MAC]: "Library/Caches",
+  [WINDOWS]: "AppData/Local/Temp",
+  [LINUX]: ".cache",
+} as const
+
+/**
+ * Function to create cache tool
+ *
+ * @example
+ * export default {
+ *   run: async (toolbox: GluegunToolbox) => {
+ *      const cache = createCacheTool(toolbox.filesystem)
+ *   }
+ * }
+ */
+export function createCacheTool(filesystem: GluegunFilesystem) {
```

Comment:
I could be missing something, so please let me know if I am.

Reply:
I think you are right about this, but it's a default habit in my brain rn

The builder is nice for doing unit tests because you can pass in a stub of tools like filesystem without having to use jest mocking, but I don't think that use case will be necessary here

Reply:
Especially since most of the methods on here are pure functions